### PR TITLE
Fail AT if exception(s) left at end of AT

### PR DESCRIPTION
**Pull request by NickGrantX**
# Motivation and Context
I have some fear that we might be able to create an AT which doesn't fail but exceptions are being thrown by RM... undetected.

# What has changed
Added a check at the end of each scenario to ensure there are no exceptions.

# How to test?
Run the ATs. Should all pass.

# Links
Trello: https://trello.com/c/SCSNxVxU

### CODE REVIEW

**Comment by NickGrantX** [comment]
[Fail ATs if there are uncaught exceptions](https://trello.com/c/hP1vE5GM/1250-fail-ats-if-there-are-uncaught-exceptions)

File: acceptance_tests/features/environment.py [hunk 1]
```diff
@@ -31,6 +32,14 @@ def before_scenario(context, _):
     purge_queues()
 
 
+def after_scenario(_, scenario):
+    if "clear_for_bad_messages" not in scenario.tags:
+        response = requests.get(f'{Config.EXCEPTION_MANAGER_URL}/badmessages')
+        response.raise_for_status()
+        if response.json():
+            test_helper.fail('Unexpected exception(s) thrown by RM')
```
Comment (AdamHawtin) [inline acceptance_tests/features/environment.py:41]: Worth logging out the response here?

Comment (NickGrantX) [reply acceptance_tests/features/environment.py:41]: the list of message hashes isn't going to be any use.

Sure we could build something to interrogate the exception manager for the details, but this is here just as a belt-and-braces check to backstop anything dodgy creeping through

**Review by ryangrundy7 — APPROVED** [review]
ATs passed 👍

**Review by AdamHawtin — APPROVED** [review]

**Pull request merged**